cmd/bucket-replication.go

```diff
@@ -316,6 +316,28 @@ func replicateDelete(ctx context.Context, dobj DeletedObjectReplicationInfo, obj
 		return
 	}
 
+	// Lock the object name before starting replication operation.
+	// Use separate lock that doesn't collide with regular objects.
+	lk := objectAPI.NewNSLock(bucket, "/[replicate]/"+dobj.ObjectName)
```

Comment:
This lock will collide with situations where some uploads this exact object name, to avoid this kind of abuse we should move the lock to `minioMetaBucket` instead

Reply:
@harshavardhana Objects cannot start with `/` which is why I choose this schema..

Reply:
Yes perhaps we can live with it..